client.go

```diff
@@ -422,11 +433,21 @@ func (client *Client) cachedMetadata(topic string, partitionID int32) *Partition
 	return nil
 }
 
-func (client *Client) cachedPartitions(topic string, partitionSet int) []int32 {
+func (client *Client) cachedPartitions(topic string, partitionSet partitionType) []int32 {
 	client.lock.RLock()
 	defer client.lock.RUnlock()
 
 	partitions := client.metadata[topic]
+
+	if partitions == nil {
+		return nil
+	}
+	return client.cachedPartitionsResults[topic][partitionSet]
+}
+
+func (client *Client) cachedPartitionsNoLock(topic string, partitionSet partitionType) []int32 {
```

Comment:
I find this function name slightly confusing; based on the name I expect `cachedPartitions` to simply call grab the lock and call this function.

Maybe `setPartitionCache` or `updatePartitionCache`?